lux/core/frame.py

```diff
@@ -305,12 +305,28 @@ def recommendation(self, recommendation: Dict):
     def current_vis(self):
         # _parse_validate_compile_intent does not call executor,
         # we only attach data to current vis when user request current_vis
-        if (
+        valid_current_vis = (
             self._current_vis is not None
             and len(self._current_vis) > 0
             and self._current_vis[0].data is None
-        ):
-            lux.config.executor.execute(self._current_vis, self)
+            and self._current_vis[0].intent
+        )
+        if valid_current_vis:
+            for clause in self._current_vis[0].intent:
+                if isinstance(clause.attribute, list):
```

Comment:
Can you explain the logic of the checks that are performed here? Is there a reason why we can't simply use the same validation logic inside the Parser and Validator?

Reply:
The check here just checks if the `clause.attribute` is "?", "Record", or an attribute in the columns of the DataFrame (both in the case that the `clause.attribute` is a String or a Python list.

I will look into using `validate_intent` to see if we can use some of the same logic here instead!